tests/lib/rules/prefer-spread.js

```diff
@@ -39,7 +39,11 @@ ruleTester.run("prefer-spread", rule, {
         // ignores incomplete things.
         "foo.apply();",
         "obj.foo.apply();",
-        "obj.foo.apply(obj, ...args)"
+        "obj.foo.apply(obj, ...args)",
+
+        // Optional chaining
+        "(a?.b).c.foo.apply(a?.b.c, args);",
+        "a?.b.c.foo.apply((a?.b).c, args);"
```

Comment:
This might be actually a false negative?

Those parens don't make much sense, but something like the following looks like a valid use of `?.` since it also skips evaluation of arguments:

```js
/* eslint prefer-spread: error */

a?.b.c.apply(a.b, args); // false negative?
```

(not sure is it worth fixing this, as it seems complex)

Reply:
If we decide to fix this, the same applies to `no-useless-call`

```js
a?.b.c.call(a.b); // false negative?

a?.b.c.apply(a.b, [1, 2, 3]);  // false negative?
```

Reply:
Currently, `prefer-spread` and `no-useless-call` use `astUtils.equalTokens()` to check if the callee and the first argument are samely or not. I think that we can use `astUtils.isSameReference()` instead. But I'm hesitating because it looks like a breaking change.

Reply:
Agreed, it seems best to not change anything now, and maybe discuss it later (after this PR).